File: backend/app/models/room.py
from sqlalchemy import String, Integer, Boolean, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.database import Base


class Room(Base):
    __tablename__ = "rooms"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(100))
    capacity: Mapped[int] = mapped_column(Integer)
    room_type: Mapped[str] = mapped_column(String(20))
    # lecture_hall | lab | outdoor
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    building_id: Mapped[int] = mapped_column(ForeignKey("buildings.id"))

    building: Mapped["Building"] = relationship(back_populates="rooms")

File: frontend/lib/core/models/room.dart
class Room {
  final int id;
  final String name;
  final int capacity;
  final String roomType;
  final bool isActive;
  final int buildingId;

  const Room({
    required this.id, required this.name, required this.capacity,
    required this.roomType, required this.isActive, required this.buildingId,
  });

  factory Room.fromJson(Map<String, dynamic> json) => Room(
        id: json['id'] as int,
        name: json['name'] as String,
        capacity: json['capacity'] as int,
        roomType: json['room_type'] as String,
        isActive: json['is_active'] as bool,
        buildingId: json['building_id'] as int,
      );
}
